src/solver/aleatoire/alea_tirage_au_sort_chroniques.cpp

```diff
@@ -185,6 +187,23 @@ static void InitializeTimeSeriesNumbers_And_ThermalClusterProductionCost(
         ptchro.TransmissionCapacities
           = (directWidth != 1) ? link->timeseriesNumbers[0][year] : 0; // zero-based
     }
+    //Each binding constraints group
+    std::vector<std::string> all_groups;
+    std::transform(runtime.bindingConstraint.begin(), runtime.bindingConstraint.end(),
+                   std::back_inserter(all_groups),
+                   [](auto& bc)->std::string { //Catch by reference else risk double delete of inner pointer members
+        return bc.group;
+    });
+    all_groups.erase(std::unique(all_groups.begin(), all_groups.end()), all_groups.end());
+    for (const auto& group_name : all_groups) {
+        for (const auto& bc: runtime.bindingConstraint)
+        {
+            if (bc.group == group_name) {
+                assert(year < bc.time_series.width); //TODO width or height ?
+            }
+        }
+        NumeroChroniquesTireesParGroup[numSpace][group_name] = study.bindingConstraints.time_series[group_name].timeseriesNumbers[0][year];
```

Comment:
You do 2 things here
- Loop on groups
- In debug mode, check consistency `year` / `bc.time_series.width`

In my opinion, it's not necessary to construct a list of groups here, you could simply loop on BCs

```suggestion
    //Each binding constraints group
    std::vector<std::string> all_groups;
    std::transform(runtime.bindingConstraint.begin(), runtime.bindingConstraint.end(),
                   std::back_inserter(all_groups),
                   [](auto& bc)->std::string { //Catch by reference else risk double delete of inner pointer members
        return bc.group;
    });
    all_groups.erase(std::unique(all_groups.begin(), all_groups.end()), all_groups.end());
    for (const auto& group_name : all_groups) {
        for (const auto& bc: runtime.bindingConstraint)
        {
            if (bc.group == group_name) {
                assert(year < bc.time_series.width); //TODO width or height ?
            }
        }
        NumeroChroniquesTireesParGroup[numSpace][group_name] = study.bindingConstraints.timeseriesNumbers(group_name, year);
```